### ISSUE TITLE
Problems with AlexNet

### ISSUE THREAD

**Issue by novelmartis** [vlfeat/matconvnet#511]
Is AlexNet supposed to work with DAG? 
This is the error which pops out:

Error using dagnn.DagNN/eval (line 80)
No variable of name 'data' could be found in the DAG.

Error in untitled3 (line 17)
net.eval({'data', im_}) ;

Also, in net.meta.normalisation, the averageimage is a 3x1 array as opposed to the "expected" 227x227x3 matrix. Is this intentional? [the dimensions don't match with im_]

Basically, I am not able to get AlexNet to work as of now [ver beta-19]

**Comment by lenck** [comment]
Hi, make sure that you have the variable names right. I think that `dagnn.DagNN.fromSimpleNN` uses either `input` or `x0` based on the [configuration](http://www.vlfeat.org/matconvnet/mfiles/+dagnn/@DagNN/DagNN/#fromsimplenn-initialize-a-dagnn-object-from-a-simplenn-network).

In case of the average image, yes some models have just `1x1x3` array. But that's fine, to adjust by the mean just use:
```
im_m = bsxfun(@minus, im, mean);
```

**Comment by novelmartis** [comment]
Thanks!

Using 'input' in place of 'data' solves the issue.

**Comment by chinnurocks** [comment]
Hi, 
I want to use ResNet and instead of using the whole network. I just wanted to use some part for which I came up with a for-loop and able to create a DAG object with name net1. I am able to run the code net.eval({'data',i}), but when I just trying to do the same with net1.eval({'data',i}). It gave me an error that data could not be found. So I replaced it with input,but still I am getting a error : 
>> net.eval({'input',i});
Error using dagnn.DagNN/eval (line 80)
No variable of name 'input' could be found in the DAG.
